src/aleph/http/compression.clj

```diff
@@ -82,6 +77,16 @@
 
 (defrecord Qvals [^double star ^double br ^double snappy ^double zstd ^double gzip ^double deflate])
 
+;; We can't reference these options classes directly in `choose-codec` since the compiler would try
+;; to initialize them even when the necessary dependencies aren't available, resulting in an
+;; error. Indirectly referencing them like this works, though.  See
+;; https://github.com/clj-commons/aleph/issues/703
+(def brotli-options-class
+  BrotliOptions)
+
+(def zstd-options-class
+  ZstdOptions)
```

Comment:
I'm surprised this works; the compiler is still looking at the whole file

Reply:
It seems to be due to how `fn` bodies are compiled. This for example also fails to compile:

```clojure
(def brotli-options-class
  (fn []
    BrotliOptions))
```

Perhaps this is a bug / unnecessary bit the compiler does when it analyzes a function body? I'll prepare a self-contained reproducer and file a question on https://ask.clojure.org/ about this.

Reply:
Hmmm, feels fragile to me. Like you're relying on internal behavior of the compiler.

The solution I originally contemplated, but figured would be a bunch of maintenance work, was to move all of these references into the Java class. That way the Clojure compiler can't blow up on it.

Reply:
https://ask.clojure.org/index.php/13810/compilation-referencing-class-failing-static-initializer :sweat_drops:

Reply:
> Hmmm, feels fragile to me. Like you're relying on internal behavior of the compiler.

Yeah, I agree with your feeling. Meanwhile [Alex replied to my question](https://ask.clojure.org/index.php/13810/compilation-referencing-class-failing-static-initializer?show=13811#a13811) and he seems to agree that it's a bug in how `fn` bodies are compiled. Makes me feel a bit easier about the workaround.

 
> The solution I originally contemplated, but figured would be a bunch of maintenance work, was to move all of these references into the Java class. That way the Clojure compiler can't blow up on it.

Perhaps this is indeed the better way to go. I'll give that a shot to see!

Reply:
OK pushed! That `instanceOf` issue mentioned in the comment derailed me for a bit there :sweat_smile: But overall this feels more robust. WDYT?